lib/vault.ts

```diff
@@ -300,38 +300,46 @@ export async function retrieveSubmissions(
  * @param email Email address of the user downloading the Submission
  */
 export async function updateLastDownloadedBy(
-  responseID: string,
+  responses: Array<{ id: string; status: string }>,
   formID: string,
-  email: string,
-  status: VaultStatus
+  email: string
 ) {
   const documentClient = connectToDynamo();
 
-  const isNewResponse = status === VaultStatus.NEW;
+  const input: TransactWriteItemsCommandInput = {
```

Comment:
This command cannot take more than a 100 operations at a time and is limited in size so you may have to have a mechanism that will split the logic into multiple requests.

This is from the [AWS documentation](https://docs.aws.amazon.com/AWSJavaScriptSDK/v3/latest/Package/-aws-sdk-client-dynamodb/Class/TransactWriteItemsCommand/):
TransactWriteItems is a synchronous write operation that groups up to 100 action requests. These actions can target items in different tables, but not in different Amazon Web Services accounts or Regions, and no two actions can target the same item. For example, you cannot both ConditionCheck and Update the same item. The aggregate size of the items in the transaction cannot exceed 4 MB.

Reply:
Refactored to process responses in chunks of 100.